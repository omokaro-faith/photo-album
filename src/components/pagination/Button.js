import React from 'react';
import { PropTypes } from 'prop-types';

const Button = ({id, handleClick, pageNumber}) => (
  <li
    id={id}
    onClick={handleClick}
  >
    {pageNumber}
  </li>
)

Button.propTypes = {
  id: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
  pageNumber: PropTypes.number.isRequired,
}



export default Button;
