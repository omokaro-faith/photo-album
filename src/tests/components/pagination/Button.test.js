import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/pagination/Button';

describe('Button', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      handleClick: jest.fn(),
      id: 1,
      pageNumber: 2
    };
    wrapper = shallow(<Button {...props}/>);
  });

  afterEach(() => { wrapper = null; });

  
  describe('renders', () => {
    it('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    it('should not have any regressions', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleClick', () => {
  it('should call props.handleClick', () => {
    const ol = wrapper.find('ol');
    ol.simulate('click');
    expect(props.handleClick).toHaveBeenCalled();
    });
  });
});
