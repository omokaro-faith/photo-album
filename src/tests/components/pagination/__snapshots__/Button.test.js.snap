// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders should not have any regressions 1`] = `ShallowWrapper {}`;
